clients/src/test/java/org/apache/kafka/common/requests/RequestTestUtils.java

```diff
@@ -208,10 +209,10 @@ public static MetadataResponse metadataUpdateWith(final String clusterId,
             for (int i = 0; i < numPartitions; i++) {
                 TopicPartition tp = new TopicPartition(topic, i);
                 Node leader = nodes.get(i % nodes.size());
-                List<Integer> replicaIds = Collections.singletonList(leader.id());
+                List<Integer> replicaIds = nodes.stream().map(Node::id).collect(Collectors.toList());
```

Comment:
Why do we need to change this? Is it OK to change for all existing usages?

Reply:
tests were failing, specifically the ones that were testing preferred read replica since it was using 2 nodes instead of 1. replica=[0,1] is expected with the new nodeIfOnline changes.

this provides the expected behavior: all nodes that are passed in are part of the replica and isr set.

Reply:
Leaving the replication factor as implicit seems less than ideal. Perhaps we could make it an explicit argument?